<launch>
  <arg name="model_name" default="double_pendulum_with_base"/>
  <arg name="event_mode" default="timer" />
  <arg name="gazebo_ns" default="/gazebo" />
  <arg name="gazebo_gui_ns" default="/gazebo_gui" />
  <group ns="randomizers">
    <node pkg="gazebo_domain_randomizer" name="light_randomizer" type="light_randomizer.py" args="-e $(arg event_mode) --gazebo_ns $(arg gazebo_ns)" output="screen" />
    <node pkg="gazebo_domain_randomizer" name="shapes_randomizer" type="shapes_randomizer.py" args="-e $(arg event_mode) --gazebo_ns $(arg gazebo_ns)" output="screen" />
    <node pkg="gazebo_domain_randomizer" name="sky_randomizer" type="sky_randomizer.py" args="-e $(arg event_mode) --gazebo_ns $(arg gazebo_gui_ns)" output="screen" />
    <node pkg="gazebo_domain_randomizer" name="link_properties_randomizer" type="link_properties_randomizer.py" args="-m $(arg model_name) -e $(arg event_mode) --gazebo_ns $(arg gazebo_ns)" output="screen" />
    <node pkg="gazebo_domain_randomizer" name="joint_properties_randomizer" type="joint_properties_randomizer.py" args="-m $(arg model_name) -e $(arg event_mode) --gazebo_ns $(arg gazebo_ns)" output="screen" />
    <node pkg="gazebo_domain_randomizer" name="physics_properties_randomizer" type="physics_properties_randomizer.py" args="-e $(arg event_mode) --gazebo_ns $(arg gazebo_ns)" output="screen" />
    <node pkg="gazebo_domain_randomizer" name="link_visual_properties_randomizer" type="link_visual_properties_randomizer.py" args="-m $(arg model_name) -e $(arg event_mode) --gazebo_ns $(arg gazebo_ns) --gazebo_gui_ns $(arg gazebo_gui_ns)" output="screen" />
    <node pkg="gazebo_domain_randomizer" name="surface_params_randomizer" type="surface_params_randomizer.py" args="-m $(arg model_name) -e $(arg event_mode) --gazebo_ns $(arg gazebo_ns)" output="screen" />
  </group>
</launch>